logic/transactions/pool.js

```diff
@@ -0,0 +1,922 @@
+		case 'recipient_pk':
+			return __private.getAllPoolTransactionsByFilter({'requesterPublicKey': params.publicKey});
+		default:
+			return 'Invalid filter';
+	}
+};
+
+/**
+ * Gets ready transactions ordered by fee and received time.
+ * @param {number} limit
+ * @return {transaction[]}
+ */
+TransactionPool.prototype.getReady = function (limit) {
+	var r = _.orderBy(pool.verified.ready.transactions, ['fee', 'receivedAt'],['desc', 'asc']);
+	if (limit && limit < r.length) {
+		r.splice(limit);
+	}
+	return r;
+};
+
+/**
+ * Checks sender has enough credit to apply transaction.
+ * @param {transaction} transaction
+ * @param {address} sender
+ * @param {function} cb - Callback function.
+ * @return {setImmediateCallback} err, transactions
+ */
+TransactionPool.prototype.checkBalance  = function (transaction, sender, cb) {
+	var poolBalance = new bignum('0');
+	var paymentTransactions;
+	var receiptTransactions;
+
+	library.logic.account.get({ address: sender.address }, 'balance', function (err, account) {
+		if (err) {
+			return setImmediate(cb, err);
+		}
+		if (account === null) {
+			account = {};
+			account.balance = 0;
+		}
+		// total payments
+		paymentTransactions = self.getAll('sender_id', { id: sender.address });
+		if (paymentTransactions.ready.length > 0) {
+			paymentTransactions.ready.forEach(function (paymentTransaction) {
+				if (paymentTransaction.amount) {
+					poolBalance = poolBalance.minus(paymentTransaction.amount.toString());
+				}
+				poolBalance = poolBalance.minus(paymentTransaction.fee.toString());
+			});
+		}
+		
+		// total receipts
+		receiptTransactions = self.getAll('recipient_id', { id: sender.address });
+		if (receiptTransactions.ready.length > 0) {
+			receiptTransactions.ready.forEach(function (receiptTransaction) {
+				if (receiptTransaction.type === transactionTypes.SEND) {
+					poolBalance = poolBalance.plus(receiptTransaction.amount.toString());
+				}
+			});
+		}
+
+		// total balance
+		var balance = new bignum(account.balance.toString());
+		balance = balance.plus(poolBalance);
+
+		// Check confirmed sender balance
+		var amount = new bignum(transaction.amount.toString()).plus(transaction.fee.toString());
+		var exceeded = balance.lessThan(amount);
+
+		if (exceeded) {
+			return setImmediate(cb, [
+				'Account does not have enough LSK:', sender.address,
+				'balance:', balance.div(Math.pow(10,8))
+			].join(' '));
+		}
+
+		return setImmediate(cb, null, [
+			'balance:', balance.div(Math.pow(10,8))
+		].join(' '));
+	});
+};
+
+/**
+ * Validates a transaction and adds it to the verified pool list.
+ * @implements {__private.addToVerified}
+ * @param {transaction} transaction
+ * @param {boolean} broadcast
+ * @param {function} cb - Callback function.
+ * @return {setImmediateCallback} error | cb
+ */
+TransactionPool.prototype.addFromPublic = function (transaction, broadcast, cb) {
+	__private.addToVerified(transaction, broadcast, cb);
```

Comment:
the public API can accept an array. 

The behavior should be:

1 tx in array - > Accept and process , return response
x > 1 tx in array -> Accept and inform the user about acceptance, but not if there are any errors

Reply:
The trigger events to add transactions are in:
`modules\loader.js`: `onPeersReady`
`modules\transport.js`: `postTransactions`
Both of them handles 1 or many transactions and at the end sends individual transactions to pool by calling `modules.transactions.processUnconfirmedTransaction`
We should change `onPeersReady` to call `modules.transactions.processPeerTransactions`